go/stellar/stellarsvc/anchor_test.go

```diff
@@ -0,0 +1,231 @@
+		if err == nil {
+			t.Errorf("err test %d [%s]: Withdraw returned no error, but expected one", i, test.Name)
+			continue
+		}
+	}
+
+	for i, test := range validAnchorTests {
+		accountID, _ := randomStellarKeypair()
+		ai := newAnchorInteractor(accountID, test.Asset)
+		ai.httpGetClient = mockAnchorUSDTransferGet
+		res, err := ai.Deposit(tc.MetaContext())
+		if err != nil {
+			t.Errorf("valid test %d [%s]: Deposit returned an error: %s", i, test.Name, err)
+			continue
+		}
+		if res.ExternalUrl == nil && res.MessageFromAnchor == nil {
+			t.Errorf("valid test %d [%s] deposit: result fields are all nil", i, test.Name)
+			continue
+		}
+		if test.DepositExternalURL != "" && res.ExternalUrl == nil {
+			t.Errorf("valid test %d [%s] deposit: result external url field is nil, expected %s", i, test.Name, test.DepositExternalURL)
+			continue
+		}
+		if res.ExternalUrl != nil {
+			if test.DepositExternalURL != *res.ExternalUrl {
+				t.Errorf("valid test %d [%s] deposit: result external url field %s, expected %s", i, test.Name, *res.ExternalUrl, test.DepositExternalURL)
+			}
+		}
+
+		res, err = ai.Withdraw(tc.MetaContext())
+		if err != nil {
+			t.Errorf("valid test %d [%s]: Withdraw returned an error: %s", i, test.Name, err)
+			continue
+		}
+		if res.ExternalUrl == nil && res.MessageFromAnchor == nil {
+			t.Errorf("valid test %d [%s] withdraw: result fields are all nil", i, test.Name)
+			continue
+		}
+		if test.WithdrawExternalURL != "" && res.ExternalUrl == nil {
+			t.Errorf("valid test %d [%s] withdraw: result external url field is nil, expected %s", i, test.Name, test.WithdrawExternalURL)
+			continue
+		}
+		if res.ExternalUrl != nil {
+			if test.WithdrawExternalURL != *res.ExternalUrl {
+				t.Errorf("valid test %d [%s] withdraw: result external url field %s, expected %s", i, test.Name, *res.ExternalUrl, test.WithdrawExternalURL)
+			}
+		}
+	}
+}
+
+// mockKeybaseTransferGet is an httpGetClient func that returns a stored result
+// for TRANSFER_SERVER/deposit and TRANSFER_SERVER/withdraw
+func mockKeybaseTransferGet(mctx libkb.MetaContext, url string) (int, []byte, error) {
+	parts := strings.Split(url, "?")
+	switch parts[0] {
+	case "https://transfer.keybase.io/transfer/deposit":
+		return http.StatusForbidden, []byte(depositBody), nil
+	case "https://transfer.keybase.io/transfer/withdraw":
+		return http.StatusForbidden, []byte(withdrawBody), nil
+	default:
+		return 0, nil, errors.New("unknown mocked url")
+	}
+
+}
+
+// mockAnchorUSDTransferGet is an httpGetClient func that returns a stored result
+// for TRANSFER_SERVER/deposit and TRANSFER_SERVER/withdraw
+func mockAnchorUSDTransferGet(mctx libkb.MetaContext, url string) (int, []byte, error) {
+	parts := strings.Split(url, "?")
+	switch parts[0] {
+	case "https://api.anchorusd.com/transfer/deposit":
+		return http.StatusForbidden, []byte(depositBody), nil
+	case "https://api.anchorusd.com/transfer/withdraw":
+		return http.StatusForbidden, []byte(withdrawBody), nil
```

Comment:
it looks like there's nothing that tests the http.StatusOK branch of logic. do we not expect these to happen / do they not actually exist in practice?

Reply:
I'll add a test, that was an oversight.  But I don't expect them to happen.

Reply:
yup. that makes sense. cool.